src/coreclr/jit/fgprofile.cpp

```diff
@@ -5063,3 +5063,115 @@ bool Compiler::fgDebugCheckOutgoingProfileData(BasicBlock* block, ProfileChecks
 }
 
 #endif // DEBUG
+
+//------------------------------------------------------------------------
+// fgRepairProfileCondToUncond: attempt to repair profile after modifying
+//   a conditinal branch to an unconditional branch.
+//
+// Arguments:
+//   block        - block that was just altered
+//   retainedEdge - flow edge that remains
+//   removedEdge  - flow edge that was removed
+//   metric       - [in/out, optional] metric to update if profile becomes inconsistent
+//
+void Compiler::fgRepairProfileCondToUncond(BasicBlock* block,
+                                           FlowEdge*   retainedEdge,
+                                           FlowEdge*   removedEdge,
+                                           int*        metric)
+{
+    assert(block->KindIs(BBJ_ALWAYS));
+    assert(block->GetTargetEdge() == retainedEdge);
+
+    // If block does not have profile data, there's nothing to do.
+    //
+    if (!block->hasProfileWeight())
+    {
+        return;
+    }
+
+    // If the removed edge was not carrying away any profile, there's nothing to do.
+    //
+    weight_t const weight = removedEdge->getLikelyWeight();
+
+    if (weight == 0)
```

Comment:
nit:
```suggestion
    if (weight == 0.0)
```
or `BB_ZERO_WEIGHT`